### PR TITLE
API/Client: Revise null-checks for output parameters 

### PR DESCRIPTION
**Pull request by wooksong**
This patch eliminates null-checks for the output parameters, which require memory allocation at the callee side.

Signed-off-by: Wook Song <wook16.song@samsung.com>

### CODE REVIEW

File: c/src/ml-api-service-agent-client.c [hunk 1]
```diff
@@ -143,9 +143,8 @@ ml_service_get_pipeline (const char *name, char **pipeline_desc)
         "The parameter, 'name' is NULL, It should be a valid string");
   }
 
-  if (!pipeline_desc) {
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "The parameter 'pipeline_desc'. It should be a valid char**");
+  if (pipeline_desc) {
+    *pipeline_desc = NULL;
```
Comment (songgot) [inline c/src/ml-api-service-agent-client.c:147]: I don't know well.
Shouldn't 'pipeline_desc' be the first address where memory is allocated?

Comment (wooksong) [reply c/src/ml-api-service-agent-client.c:147]: This is a **[out]** parameter.

An example usage as follows:
```
char *desc;

ml_service_get_pipeline(name, &desc);
```

```desc``` will point the first block of an allocated memory inside wherever ```ml_service_get_pipeline```.

**Review by songgot — APPROVED** [review]
LGTM

File: c/src/ml-api-service-agent-client.c (hunk 1 above)
Comment (songgot) [reply c/src/ml-api-service-agent-client.c:147]: Thank you for the detailed description.

File: c/src/ml-api-service-agent-client.c [hunk 2]
```diff
@@ -318,6 +339,7 @@ ml_service_get_pipeline_state (ml_service_h h, ml_pipeline_state_e * state)
   g_clear_error (&err);
 
   *state = (ml_pipeline_state_e) _state;
+
```
Comment (myungjoo) [inline c/src/ml-api-service-agent-client.c:342]: nah..

**Review by myungjoo — APPROVED** [review]
LGTM, except for the empty line add.

File: c/src/ml-api-service-agent-client.c (hunk 2 above)
Comment (wooksong) [reply c/src/ml-api-service-agent-client.c]: Removed. Thanks.

**Review by niley7464 — APPROVED** [review]
LGTM 👍

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**